src/DotVVM.Framework/Resources/Scripts/spa/spa.ts

```diff
@@ -8,27 +8,28 @@ import { DotvvmPostbackError } from '../shared-classes';
 export const isSpaReady = ko.observable(false);
 
 export function init(): void {
-    const spaPlaceHolder = getSpaPlaceHolder();
-    if (spaPlaceHolder == null) {
-        throw new Error("The SpaContentPlaceHolder control was not found!");
+    const spaPlaceHolders = getSpaPlaceHolders();
+    if (!spaPlaceHolders) {
+        throw new Error("No SpaContentPlaceHolder control was found!");
     }
 
-    window.addEventListener("hashchange", event => handleHashChangeWithHistory(spaPlaceHolder, false));
-    handleHashChangeWithHistory(spaPlaceHolder, true);
+    window.addEventListener("hashchange", event => handleHashChangeWithHistory(spaPlaceHolders, false));
+    handleHashChangeWithHistory(spaPlaceHolders, true);
 
-    window.addEventListener('popstate', event => handlePopState(event, spaPlaceHolder != null));
+    window.addEventListener('popstate', event => handlePopState(event, spaPlaceHolders[0] != null));
 }
 
-function getSpaPlaceHolder(): HTMLElement | null {
-    const elements = document.getElementsByName("__dot_SpaContentPlaceHolder");
-    if (elements.length == 1) {
-        return <HTMLElement> elements[0];
-    }
-    return null;
+function getSpaPlaceHolders(): NodeListOf<HTMLElement> | null {
+    return document.getElementsByName("__dot_SpaContentPlaceHolder");
 }
 
-export function getSpaPlaceHolderUniqueId(): string {
-    return getSpaPlaceHolder()!.getAttribute("data-dotvvm-spacontentplaceholder")!;
+export function getSpaPlaceHoldersUniqueId(): string {  
+    let identifier = "";
+    getSpaPlaceHolders()!.forEach(function (element) {
```

Comment:
```suggestion
    getSpaPlaceHolders()!.forEach((element) => {
```

We generally prefer arrow functions, it is shorter and behaves more predictably when it comes to `this`

Reply:
OK. Will be using arrow function from now on.